packages/cli/src/modules/start/commands/repo/start.ts

```diff
@@ -0,0 +1,216 @@
+    const matchingPackages = packages.filter(
+      pkg => pkg.packageJson.backstage?.role === role,
+    );
+    if (matchingPackages.length > 1) {
+      // Final fallback is to check for the package path within the monorepo, packages/app or packages/backend
+      const expectedPath = paths.resolveTargetRoot(
+        role === 'frontend' ? 'packages/app' : 'packages/backend',
+      );
+      const matchByPath = matchingPackages.find(
+        pkg => relativePath(expectedPath, pkg.dir) === '',
+      );
+      if (matchByPath) {
+        targetPackages.push(matchByPath);
+        continue;
+      }
+
+      throw new Error(
+        `Found multiple packages with role '${role}' but none of the use the default path '${expectedPath}',` +
+          `choose which packages you want to run by passing the package names explicitly ` +
+          `as arguments, for example 'yarn backstage-cli repo start my-app my-backend'.`,
+      );
+    }
+
+    targetPackages.push(...matchingPackages);
+  }
+  if (targetPackages.length > 0) {
+    return targetPackages;
+  }
+
+  // If no app or backend packages are found, fall back to expecting single plugin packages
+  for (const role of ['frontend-plugin', 'backend-plugin']) {
+    const matchingPackages = packages.filter(
+      pkg => pkg.packageJson.backstage?.role === role,
+    );
+    if (matchingPackages.length > 1) {
+      throw new Error(
+        `Found multiple packages with role '${role}', please choose which packages you want` +
+          `to run by passing the package names explicitly as arguments, for example ` +
+          `'yarn backstage-cli repo start my-plugin my-plugin-backend'.`,
+      );
+    }
+    targetPackages.push(...matchingPackages);
+  }
+  if (targetPackages.length > 0) {
+    return targetPackages;
+  }
+
+  throw new Error(
+    `Unable to find any packages with role 'frontend', 'backend', 'frontend-plugin', or 'backend-plugin'.`,
+  );
+}
+
+async function resolvePackageOptions(
+  targetPackages: BackstagePackage[],
+  options: CommandOptions,
+) {
+  const linkedWorkspace = await resolveLinkedWorkspace(options.link);
+
+  return targetPackages.flatMap(pkg => {
+    const startScript = pkg.packageJson.scripts?.start;
+    if (!startScript) {
+      console.log(
+        `No start script found for package ${pkg.packageJson.name}, skipping...`,
+      );
+      return [];
+    }
+
+    // Grab and parse --config and --require options from the start scripts, the rest are ignored
+    // TODO(Rugvip): Prolly switch over to completely different arg parsing to avoid this duplication
+    const { values: parsedOpts } = parseArgs({
+      args: startScript.split(' '),
+      strict: false,
+      options: {
+        config: {
+          type: 'string',
+          multiple: true,
+        },
+        require: {
+          type: 'string',
+        },
+      },
+    });
+    const parsedRequire =
+      typeof parsedOpts.require === 'string' ? parsedOpts.require : undefined;
```

Comment:
what about supporting multiple require?

Reply:
it's not supported by package start atm

Reply:
leaving this for a followup to add support for it both here and in the existing `package start` command